other/docker/windows/build_toxcore.sh

```diff
@@ -109,6 +111,22 @@ build() {
     fi
   fi
 
+  # generate def, lib and exp as they supposedly help with linking against the dlls,
+  # especially the lib is supposed to be of great help when linking on msvc.
+  # cd in order to keep the object names inside .lib and .dll.a short
+  cd "$RESULT_PREFIX_DIR"/bin/
+  for TOX_DLL in *.dll; do
+    gendef - "$TOX_DLL" > "${TOX_DLL%.*}.def"
```

Comment:
It's also possible to ask CMake to create this: https://cmake.org/cmake/help/v3.29/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html

Not sure if it's better or worse, though.

Reply:
In this case they should do the same, but we should either generate the .def using the public headers + hardcoded magic, or decorate all function calls in the .h with dllexport.

Reply:
@robinlinden see https://github.com/TokTok/c-toxcore/pull/2713#issuecomment-1967684863 as GitHub wouldn't allow me to post a message so big as a review comment, kept getting HTTP 500 from the comment endpoint with no visible error message.